contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+    }
+
+    /**
+     * @notice sets the fill deadline duration
+     * @dev can only be called by owner
+     * @param _fillDeadline fill deadline duration
+     */
+    function setFillDeadline(uint256 _fillDeadline) external onlyOwner {
+        require(_fillDeadline > 0, "OTCWrapper: fill deadline cannot be 0");
+
+        fillDeadline = _fillDeadline;
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice allows market maker to deposit collateral
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID id of the order
+     * @param _amount amount to deposit
+     */
+    function depositCollateral(uint256 _orderID, uint256 _amount) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.seller == msg.sender, "OTCWrapper: sender is not the order seller");
+
+        IERC20 collateralInterface = IERC20(order.collateral);
+
+        // market maker inflow - an approve() by the msg.sender is required beforehand
+        collateralInterface.safeTransferFrom(msg.sender, address(this), _amount);
+
+        // approve margin pool to deposit collateral
+        collateralInterface.safeApproveNonCompliant(addressbook.getMarginPool(), _amount);
+
+        UtilsWrapperInterface.ActionArgs[] memory actions = new UtilsWrapperInterface.ActionArgs[](1);
+
+        actions[0] = UtilsWrapperInterface.ActionArgs(
+            UtilsWrapperInterface.ActionType.DepositCollateral,
+            address(this), // owner
+            address(this), // address to transfer from
+            order.collateral, // deposited asset
+            order.vaultID, // vaultId
+            _amount, // amount
+            0, //index
+            "" //data
+        );
+
+        // execute actions
+        controller.operate(actions);
+
+        emit CollateralDeposited(_orderID, _amount);
+    }
+
+    /**
+     * @notice allows market maker to withdraw collateral
+     * @dev can only be called by the market maker who is the order seller
+     * @param _orderID id of the order
+     * @param _amount amount to withdraw
+     */
+    function withdrawCollateral(uint256 _orderID, uint256 _amount) external nonReentrant {
+        require(_orderID <= latestOrder, "OTCWrapper: inexistent order");
+
+        Order memory order = orders[_orderID];
+
+        require(order.seller == msg.sender, "OTCWrapper: sender is not the order seller");
+
+        (UtilsWrapperInterface.Vault memory vault, , ) = controller.getVaultWithDetails(address(this), order.vaultID);
+
+        require(
+            marginRequirements.checkWithdrawCollateral(
+                order.seller,
+                order.notional,
+                _amount,
+                order.oToken,
+                order.vaultID,
+                vault
```

Comment:
we only need to pass in collateralAmount currently in the vault (instead of `vault` itself) . The collateral asset we already know (referring to https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/files#r1136432582)

Reply:
Same as this [comment ](https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2#discussion_r1136810121)